Code/GraphMol/ChemReactions/ReactionRunner.cpp

```diff
@@ -162,17 +162,19 @@ const Atom *findHighestCIPNeighbor(const Atom *atom, const Atom *skipAtom) {
     }
     unsigned cip = 0;
     if (!neighbor->getPropIfPresent(common_properties::_CIPRank, cip)) {
-      // If at least one of the atoms doesn't have a CIP rank, the highest rank
-      // does not make sense, so return a nullptr.
-      return nullptr;
-    }
-    if (cip > bestCipRank || bestCipRankedAtom == nullptr) {
+      if (bestCipRankedAtom == nullptr) {
+        bestCipRankedAtom = neighbor;
+      }
```

Comment:
I was wondering about this change.

When I wrote that, I was thinking that, since this is used only for Z/E bonds, which rely on CIP rankings, finding one atom without a rank is a signal that something had happened to the molecule, and that the Z/E labels might not be accurate anymore.

Reply:
Huh, that diff came out weird...

Reply:
The problem with the change as is was that it was causing PRECONDTION exceptions for reactions where the bonds are labeled E/Z but atoms don’t have the CIPRank property. This can unfortunately happen easily when molecules are serialized and then deserialized

Reply:
The problem with the change as is was that it was causing PRECONDTION exceptions for reactions where the bonds are labeled E/Z but atoms don’t have the CIPRank property. This can unfortunately happen easily when molecules are serialized and then deserialized

Reply:
That's unfortunate. But, still, I'm not sure that is the behavior we want, since we would just be treating the first neighbor we see as if it was the highest CIP ranked one, and that might not be the case. By doing that and running the reaction, we might just propagate a stereo that might be right or not.

I would rather prefer to fix the serialization/deserialization problem, if that is your concern (e.g. by serializing the bond as if it was cis/trans, if that one works right), or re-rank the molecule before the reaction, or just stripping the stereo from the bond if we can't trust it.

Reply:
I wonder if it would be safer to catch this at an outer scope. Maybe we could return an empty vector, and `forwardReactantBondStereo()` could abort if there are no stereo atoms?

Reply:
I think I found a way out of this that allows older serialized molecules to still be used but doesn't introduce the risk of creating bogus stereo: In the rare cases where there are neither stereoatoms nor CIP ranks present just remove the stereo from that product bond.
I'll update the PR

Reply:
This makes sense to me.